backend/btrixcloud/crawlconfigs.py

```diff
@@ -1578,29 +1579,44 @@ async def stats_recompute_all(
             last_crawl = res
             last_crawl_size = crawl_size
 
-        # only update last_crawl if no crawls running, otherwise
-        # lastCrawl* stats are already for running crawl
-        running_crawl = await crawl_config_ops.get_running_crawl(cid)
-
-        if last_crawl and not running_crawl:
-            update_query["totalSize"] = total_size
-            update_query["crawlSuccessfulCount"] = successful_count
-
-            update_query["lastCrawlId"] = str(last_crawl.get("_id"))
-            update_query["lastCrawlStartTime"] = last_crawl.get("started")
-            update_query["lastStartedBy"] = last_crawl.get("userid")
-            update_query["lastStartedByName"] = last_crawl.get("userName")
-            update_query["lastCrawlState"] = last_crawl.get("state")
-            update_query["lastCrawlSize"] = last_crawl_size
-            update_query["lastCrawlStats"] = last_crawl.get("stats")
-            update_query["lastCrawlStopping"] = False
-            update_query["isCrawlRunning"] = False
-
-            last_crawl_finished = last_crawl.get("finished")
-            update_query["lastCrawlTime"] = last_crawl_finished
-
-            if last_crawl_finished:
-                update_query["lastRun"] = last_crawl_finished
+    # always update these
+    update_query["crawlSuccessfulCount"] = successful_count
+    update_query["totalSize"] = total_size
+
+    # only update last_crawl if no crawls running, otherwise
+    # lastCrawl* stats are already for running crawl
+    running_crawl = await crawl_config_ops.get_running_crawl(cid)
+
+    if last_crawl and not running_crawl:
+        update_query["lastCrawlId"] = str(last_crawl.get("_id"))
+        update_query["lastCrawlStartTime"] = last_crawl.get("started")
+        update_query["lastStartedBy"] = last_crawl.get("userid")
+        update_query["lastStartedByName"] = last_crawl.get("userName")
+        update_query["lastCrawlState"] = last_crawl.get("state")
+        update_query["lastCrawlSize"] = last_crawl_size
+        update_query["lastCrawlStats"] = last_crawl.get("stats")
+        update_query["lastCrawlStopping"] = False
+        update_query["isCrawlRunning"] = False
+
+        last_crawl_finished = last_crawl.get("finished")
+        update_query["lastCrawlTime"] = last_crawl_finished
+
+        if last_crawl_finished:
+            update_query["lastRun"] = last_crawl_finished
+
+    elif not last_crawl:
```

Comment:
Just popping in briefly but I think we might want this to be `elif not last_crawl and not running crawl:` or else the running crawl stats might be overwritten

Reply:
good point, updated!